pynest/examples/Potjans_2014/network.py

```diff
@@ -0,0 +1,475 @@
+                nest.SetStatus(poisson, {'rate': rate_ext[i]})
+                self.poisson.append(poisson)
+
+    def create_dc_generator(self):
+        """ This function creates a DC input generator.
+
+        If DC input is provided, the DC generators are created and the
+        necessary parameters are passed to them.
+
+        """
+        if self.stim_dict['dc_input']:
+            print('DC generator created')
+            dc_amp_stim = self.net_dict['K_ext'] * self.stim_dict['dc_amp']
+            self.dc = []
+            print('DC_amp_stim', dc_amp_stim)
+            for i, target_pop in enumerate(self.pops):
+                dc = nest.Create(
+                    'dc_generator', params={
+                        'amplitude': dc_amp_stim[i],
+                        'start': self.stim_dict['dc_start'],
+                        'stop': (
+                            self.stim_dict['dc_start'] +
+                            self.stim_dict['dc_dur']
+                            )
+                        }
+                    )
+                self.dc.append(dc)
+
+    def create_connections(self):
+        """ This function creates the recurrent connections.
+
+        The recurrent connections between the neuronal populations are created.
+
+        """
+        print('Recurrent connections are established')
+        mean_delays = self.net_dict['mean_delay_matrix']
+        std_delays = self.net_dict['std_delay_matrix']
+        for i, target_pop in enumerate(self.pops):
+            for j, source_pop in enumerate(self.pops):
+                synapse_nr = int(self.synapses_scaled[i][j])
+                if synapse_nr >= 0.:
+                    weight = self.weight_mat[i][j]
+                    w_sd = abs(weight * self.weight_mat_std[i][j])
+                    conn_dict_rec = {
+                        'rule': 'fixed_total_number', 'N': synapse_nr
+                        }
+                    syn_dict = {
+                        'model': 'static_synapse',
+                        'weight': {
+                            'distribution': 'normal_clipped', 'mu': weight,
+                            'sigma': w_sd
+                            },
+                        'delay': {
+                            'distribution': 'normal_clipped',
+                            'mu': mean_delays[i][j], 'sigma': std_delays[i][j],
+                            'low': self.sim_resolution
+                            }
+                        }
+                    if weight < 0:
+                        syn_dict['weight']['high'] = 0.0
+                    else:
+                        syn_dict['weight']['low'] = 0.0
+                    nest.Connect(
+                        source_pop, target_pop,
+                        conn_spec=conn_dict_rec,
+                        syn_spec=syn_dict
+                        )
+
+    def connect_poisson(self):
+        """ Connects the Poisson generator to the microcircuit."""
```

Comment:
change to "Poisson generators" to avoid the impression that all neurons share one generator.